### PR TITLE
Second factor revocation email

### PR DESCRIPTION
**Pull request by arothuis**
[100800338](https://www.pivotaltracker.com/story/show/100800338)

This adds sending a configurable email upon revocation of a second factor registration.

### CODE REVIEW

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/Service/SecondFactorMailService.php [hunk 1]
```diff
@@ -308,4 +313,117 @@ public function sendVettedEmail(
...
+    }
+
+    /**
+     * @param Locale $locale
+     * @param CommonName $commonName
+     * @param Email $email
+     * @param SecondFactorType $secondFactorType
+     * @param SecondFactorIdentifier $secondFactorIdentifier
+     */
+    public function sendVettedSecondFactorRevokedBySelfEmail(
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/Service/SecondFactorMailService.php]: What about `Self` -> `Registrant`

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/DependencyInjection/SurfnetStepupMiddlewareCommandHandlingExtension.php [hunk 2]
```diff
@@ -49,6 +49,6 @@ public function load(array $config, ContainerBuilder $container)
             ->getDefinition('surfnet_stepup_middleware_command_handling.service.second_factor_mail')
             ->replaceArgument(4, $config['self_service_email_verification_url_template'])
             ->replaceArgument(6, $config['email_fallback_locale'])
-            ->replaceArgument(8, $config['warn_on_missing_email_template']);
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/DependencyInjection/SurfnetStepupMiddlewareCommandHandlingExtension.php]: This wasn't used? :')

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Processor/EmailProcessor.php [hunk 3]
```diff
@@ -20,18 +20,24 @@
...
 use Surfnet\Stepup\Identity\Event\YubikeyPossessionProvenEvent;
 use Surfnet\StepupMiddleware\ApiBundle\Configuration\Service\InstitutionConfigurationOptionsService;
 use Surfnet\StepupMiddleware\ApiBundle\Configuration\Service\RaLocationService;
+use Surfnet\StepupMiddleware\ApiBundle\Identity\Service\IdentityService;
 use Surfnet\StepupMiddleware\ApiBundle\Identity\Service\RaListingService;
 use Surfnet\StepupMiddleware\ApiBundle\Identity\Value\RegistrationAuthorityCredentials;
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\Service\SecondFactorMailService;
 
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects) Uses several services for information needed for relevant emails
```
Comment (rjkip) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Processor/EmailProcessor.php]: Sounds (and looks) like we ought to split this processor up into multiple.

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/Service/SecondFactorMailService.php (hunk 1 above)
Comment (arothuis) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/Service/SecondFactorMailService.php]: Agreed.

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Processor/EmailProcessor.php (hunk 3 above)
Comment (arothuis) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Processor/EmailProcessor.php]: Sounds like a plan! We could group them by their responsibility within the system: Registration, Vetting, SecondFactorRevocation.

**Comment by arothuis** [comment]
Splitting the mail service into separate services still needs some work: revocation mails are not coming through.

**Pull request merged**